File: src/app/models/PaymentCard.js

// Define collection and schema for PaymentCard
export interface PaymentCard {
    cardNumber:
	type : CardPAN
    embossedName:
	type : string
    expiryMonth:
	type : number
    expiryYear:
	type : number
    Bank:
	type : Schema.Types.ObjectId
    Account:
	type : Schema.Types.ObjectId
    Customer:
	type : Schema.Types.ObjectId
    Transactions:
 	type : [{ type: Schema.Types.ObjectId, ref: 'Transaction' }]
    CardType:
 	type : String
    CardStatus:
 	type : String
    Network:
 	type : String
#
    collection: 'paymentCards'
}

File: bankingOnGolang/internal/model/PaymentCard.go

package model

import (
)

//==============================================================
// PaymentCard Declaration
//==============================================================
type PaymentCard struct {
    BaseModel
     CardNumber            CardPAN
    EmbossedName            string
    ExpiryMonth            int32
    ExpiryYear            int32
    BankId         *uint
    Bank           *Bank `gorm:"foreignKey:BankId"`
    AccountId         *uint
    Account           *Account `gorm:"foreignKey:AccountId"`
    CustomerId         *uint
    Customer           *Customer `gorm:"foreignKey:CustomerId"`
     Transactions           []Transaction `gorm:"foreignKey:TransactionsFromPaymentCardId"`
    CardType            CardType
    CardStatus            CardStatus
    Network            CardNetwork

// parent associations as their child

}

